src/main/groovy/org/jetbrains/grammarkit/GrammarKit.groovy

```diff
@@ -8,27 +8,51 @@ class GrammarKit implements Plugin<Project> {
     @Override
     void apply(Project target) {
         def grammarKitExtension = target.extensions.create("grammarKit", GrammarKitPluginExtension.class)
+        target.configurations {
+          grammarKitClassPath
+        }
         target.afterEvaluate {
             target.repositories {
                 maven { url "https://cache-redirector.jetbrains.com/intellij-dependencies" }
+                maven { url "https://www.jetbrains.com/intellij-repository/releases" }
                 maven { url 'https://www.jitpack.io' }
             }
             target.dependencies.add(
-                    JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME,
-                    "com.github.JetBrains:Grammar-Kit:${grammarKitExtension.grammarKitRelease}",
-                    {
-                        exclude group: 'org.jetbrains.plugins'
-                        exclude module: 'idea'
-                    })
+                    target.configurations.grammarKitClassPath.name,
+                    "com.github.JetBrains:Grammar-Kit:${grammarKitExtension.grammarKitRelease}"
+            )
+            target.dependencies.add(
+                    target.configurations.grammarKitClassPath.name,
+                    "org.jetbrains.intellij.deps.jflex:jflex:${grammarKitExtension.jflexRelease}"
+            )
             target.dependencies.add(
```

Comment:
from here you are kinda enforcing your approach. But not everyone want to use it.